src/main/java/com/datatorrent/netlet/AbstractClient.java

```diff
@@ -49,7 +49,7 @@
 
   public boolean isConnected()
   {
-    return key.isValid() && ((SocketChannel)key.channel()).isConnected();
+    return key != null && key.isValid() && ((SocketChannel)key.channel()).isConnected();
```

Comment:
In case `isConnected()` is called for the AbstractClient that did not connect successfully as in 'AbstractClientTest.testUnresolvedException()'.